tests/unit/clients/test_gdns.py

```diff
@@ -101,10 +91,18 @@ def client(mocker):
         mocked.get(url, status=200)
         # paginated requests
         mocked.get(url, status=200)
-        records = await client.get_records_for_zone('a-zone')
+        records = await client.get_records_for_zone('a.zone.')
 
         assert all(
             [isinstance(r, gdns.GCPResourceRecordSet) for r in records])
         assert 6 == len(records)
 
     assert 1 == len(caplog.records)
+
+
+@pytest.mark.parametrize('dns_zone,exp_managed_zone', [
+    ('example.com.', 'example-com'),
+    ('30.20.10.in-addr.arpa.', 'reverse-20-10')
```

Comment:
Would you prefer the client raise an error, or what do you expect should happen in that case? 
I like that you brought it up, because I hadn't thought about it but I also like the idea of keeping the client as dumb as possible, and allow errors to pass through and leave the policing of what is or is not a valid zone to the plugins. In that case no error would be raised, but I wouldn't necessarily want to give an example of using the client that way by adding a test for it.

Reply:
Yeah I guess we should leave it.